### PR TITLE
[FLINK-23287][docs][table] Create user document for Window Join in SQL

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change

The pr adds documentation for Window join.

## Brief change log

Window-join.md

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by beyond1920** [comment]
@wuchong Thanks for review, I've updated. About `Put it after "Join" page? It's strange to put it between "TopN" and "Window TopN".`, could you please tell me how to move?

**Comment by beyond1920** [comment]
@wuchong Thanks for review, I've updated. Please have a look. Thanks a lot.

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+## SEMI
+Semi Window Joins returns a row from one left record if there is at least one matching row on the right side within the common window.
+
+```sql
+Flink SQL> SELECT *
+           FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE LeftTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
+           ) L WHERE L.Num IN (
+             SELECT Num FROM (   
+               SELECT * FROM TABLE(TUMBLE(TABLE RightTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:120]: row_time -> `Time` . 


Why not using a lower-case and non-reserved-keyword as the field names...?

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
++------------------+-------+------+------------------+------------------+-------------------------+
+| 2020-04-15 12:03 |  3    |  L3  | 2020-04-15 12:00 | 2020-04-15 12:05 | 2020-04-15 12:04:59.999 |
++------------------+-------+------+------------------+------------------+-------------------------+
+
+Flink SQL> SELECT *
+           FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE LeftTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
+           ) L WHERE EXISTS (
+             SELECT * FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE RightTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:133]: row_time -> Time .

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+## ANTI
+###### Anti Window Joins are the obverse of the Inner Window Join: they contain all of the unjoined rows within each common window.
+
+```sql
+Flink SQL> SELECT *
+           FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE LeftTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
+           ) L WHERE L.Num NOT IN (
+             SELECT Num FROM (   
+               SELECT * FROM TABLE(TUMBLE(TABLE RightTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:154]: row_time -> Time .

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+| 2020-04-15 12:02 |  1    |  L1  | 2020-04-15 12:00 | 2020-04-15 12:05 | 2020-04-15 12:04:59.999 |
+| 2020-04-15 12:06 |  2    |  L2  | 2020-04-15 12:05 | 2020-04-15 12:10 | 2020-04-15 12:09:59.999 |
++------------------+-------+------+------------------+------------------+-------------------------+
+
+Flink SQL> SELECT *
+           FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE LeftTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
+           ) L WHERE NOT EXISTS (
+             SELECT * FROM (
+               SELECT * FROM TABLE(TUMBLE(TABLE RightTable, DESCRIPTOR(row_time), INTERVAL '5' MINUTES))
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:168]: row_time -> Time .

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+The syntax of INNER/LEFT/RIGHT/FULL OUTER WINDOW JOIN are very similar with each other, we only give an example for FULL OUTER JOIN here.
+When performing a window join, all elements with a common key and a common tumbling window are joined together. We only give an example for a Window Join which works on a Tumble Window TVF.
+By scoping the region of time for the join into fixed five-minute intervals, we chopped our datasets into two distinct windows of time: [12:00, 12:05) and [12:05, 12:10). The L2 and R2 rows could not join together because they fell into separate windows.
+
+```sql
+Flink SQL> desc LeftTable;
++-------------+------------------------+------+-----+--------+---------------------------------+
+|        name |                   type | null | key | extras |                       watermark |
++-------------+------------------------+------+-----+--------+---------------------------------+
+|     Time  | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:57]:
```suggestion
|     Time   | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+| 2020-04-15 12:02 |  1    |  L1  |
+| 2020-04-15 12:06 |  2    |  L2  |
+| 2020-04-15 12:03 |  3    |  L3  |
++------------------+-------+------+
+
+Flink SQL> desc RightTable;
++-------------+------------------------+------+-----+--------+---------------------------------+
+|        name |                   type | null | key | extras |                       watermark |
++-------------+------------------------+------+-----+--------+---------------------------------+
+|     Time  | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:75]:
```suggestion
|     Time   | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```

**Comment by beyond1920** [comment]
@wuchong Jark, I've updated the input schema to use a lower-case and non-reserved-keyword as the field names.

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+The syntax of INNER/LEFT/RIGHT/FULL OUTER WINDOW JOIN are very similar with each other, we only give an example for FULL OUTER JOIN here.
+When performing a window join, all elements with a common key and a common tumbling window are joined together. We only give an example for a Window Join which works on a Tumble Window TVF.
+By scoping the region of time for the join into fixed five-minute intervals, we chopped our datasets into two distinct windows of time: [12:00, 12:05) and [12:05, 12:10). The L2 and R2 rows could not join together because they fell into separate windows.
+
+```sql
+Flink SQL> desc LeftTable;
++-------------+------------------------+------+-----+--------+---------------------------------+
+|        name |                   type | null | key | extras |                       watermark |
++-------------+------------------------+------+-----+--------+---------------------------------+
+| row_time    | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:57]:
```suggestion
| row_time    | TIMESTAMP(3) *ROWTIME* | true |     |        |    rowtime - INTERVAL '1' SECOND |
```

File: docs/content.zh/docs/dev/table/sql/queries/window-join.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+| 2020-04-15 12:02 |  1    |  L1  |
+| 2020-04-15 12:06 |  2    |  L2  |
+| 2020-04-15 12:03 |  3    |  L3  |
++------------------+-------+------+
+
+Flink SQL> desc RightTable;
++-------------+------------------------+------+-----+--------+---------------------------------+
+|        name |                   type | null | key | extras |                       watermark |
++-------------+------------------------+------+-----+--------+---------------------------------+
+| row_time    | TIMESTAMP(3) *ROWTIME* | true |     |        |    `Time` - INTERVAL '1' SECOND |
```
Comment (wuchong) [inline docs/content.zh/docs/dev/table/sql/queries/window-join.md:75]:
```suggestion
| row_time    | TIMESTAMP(3) *ROWTIME* | true |     |        |    row_time - INTERVAL '1' SECOND |
```

**Review by wuchong — COMMENTED** [review]
Usually, we suggest to copy queries into SQL CLI and copy outputs to docs when writing examples.

**Comment by beyond1920** [comment]
@wuchong Thanks for suggestion, I've updated the results of sql query.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**